test.gd

```diff
@@ -3,33 +3,189 @@ extends SceneTree
 
-func benchmark():
+func benchmark_raw():
   print('Benchmarking ...')
 
-  var begin = OS.get_ticks_msec()
-  var index = 0
+  var uuids := []
+  var begin = Time.get_unix_time_from_system()
 
-  while index < NUMBER_OF_TEST:
+  for i in range(NUMBER_OF_TESTS):
     uuid_util.v4()
-    index += 1
+    
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   NUMBER_OF_TESTS / duration,
+   (duration / NUMBER_OF_TESTS) * 1000000,
+   duration
+  ])
+  print('Benchmark done')
+  
+func benchmark_raw_rng():
+  print('Benchmarking ...')
+
+  var rng = RandomNumberGenerator.new()
+  var uuids := []
+  var begin = Time.get_unix_time_from_system()
+
+  for i in range(NUMBER_OF_TESTS):
+    uuid_util.v4_rng(rng)
+    
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   NUMBER_OF_TESTS / duration,
+   (duration / NUMBER_OF_TESTS) * 1000000,
+   duration
+  ])
+  print('Benchmark done')
+  
+func benchmark_obj():
+  print('Benchmarking ...')
+
+  var begin = Time.get_unix_time_from_system()
+
+  for i in range(NUMBER_OF_TESTS):
+    uuid_util.new().free()  # immediately freeing does not seem to add much overhead
+
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   NUMBER_OF_TESTS / duration,
+   (duration / NUMBER_OF_TESTS) * 1000000,
+   duration
+  ])
+  print('Benchmark done')
+  
+func benchmark_obj_rng():
+  print('Benchmarking ...')
+
+  var rng = RandomNumberGenerator.new()
+  var begin = Time.get_unix_time_from_system()
 
-  var duration = 1.0 * OS.get_ticks_msec() - begin
+  for i in range(NUMBER_OF_TESTS):
+    uuid_util.new(rng).free()  # immediately freeing does not seem to add much overhead
 
-  print('uuid/sec: %.02f (time: %sms)' % [ NUMBER_OF_TEST / duration, duration])
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   NUMBER_OF_TESTS / duration,
+   (duration / NUMBER_OF_TESTS) * 1000000,
+   duration
+  ])
+  print('Benchmark done')
+  
+func benchmark_obj_to_dict():
+  print('Setting up benchmark ...')
+  var uuids = []
+  for i in range(NUMBER_OF_OBJECTS):
+    uuids.append(uuid_util.new())
+
+  print('Benchmarking ...')
+  var begin = Time.get_unix_time_from_system()
+
+  for uuid in uuids:
+    uuid.as_dict()
+
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   NUMBER_OF_TESTS / duration,
+   (duration / NUMBER_OF_TESTS) * 1000000,
+   duration
+  ])
+  print('Cleaning up ...')
+  for uuid in uuids:
+    uuid.free()
+  print('Benchmark done')
+  
+func benchmark_obj_to_str():
+  print('Setting up benchmark ...')
+  var uuids = []
+  for i in range(NUMBER_OF_OBJECTS):
```

Comment:
Same applies here. Replace the `for` function with a `while` please.

Reply:
Here I don't think replacing `for` with `while` would be beneficial. Here the amount of memory used by `UUID` objects grows much faster than memory use of `range()`, meaning that you'd run out of memory before `range()` would cause a crash. Correct me if I'm wrong though.

Reply:
I think you're right. Still if you're benchmarking it's better to use the most optimized piece of code nonetheless.